setup_wizard/character_rig_setup/npc_rig_script.py

```diff
@@ -2109,8 +2109,10 @@ def swap_const_follow_in_const(bone, constraint_type, new_var):
     if is_version_4:
         armature = bpy.context.object.data
         collections = armature.collections
-        for coll in collections:
-            collections.remove(coll)
+
+        del_collections = bpy.context.active_object.data.collections
+        while del_collections:
+            del_collections.remove(del_collections[0])

```

Comment:
Is the while-loop an important part of the fix? or is it more of the `bpy.context.active_object.data.collections` that is the fix?

I'm curious if the original for-loop would have technically worked too or if modifying the `collections` in the for-loop threw an error

Reply:
copy pasted in case you didn't see it lol: 
Check this out, this is why we needed the change. https://projects.blender.org/blender/blender/issues/124891
Our original loop loops through the collection as we're actively removing from it. The blender developer that responded confirmed that was unintended behavior. In previous version of Blender, we got away with it scot-free, but with 4.1 it crashes. As they pointed out, a .clear() method is ideal but we don't have that yet lmao. The solution was this response here: https://projects.blender.org/blender/blender/issues/124891#issuecomment-1245473
Not sure why the OP fought with the Blender dev after he got his fix but yeah lol.

Reply:
Thanks, I saw your initial comment but forgot to reply.

Reading through this, yeah, this makes sense. The root cause that they reported was initially a thought in the back of my mind (iterating through a collection while modifying it) when I was reviewing your changes.